packages/gatsby/src/bootstrap/load-themes/index.ts

```diff
@@ -106,16 +141,29 @@ const processTheme = (
     )
   } else {
     // if a theme doesn't define additional themes, return the original theme
-    return [{ themeName, themeConfig, themeSpec, themeDir, parentDir: rootDir }]
+    return Promise.resolve([
+      { themeName, themeConfig, themeSpec, themeDir, parentDir: rootDir },
+    ])
   }
 }
 
-module.exports = async (
-  config,
-  { useLegacyThemes = false, configFilePath, rootDir }
-) => {
+interface ILoadThemesOptions {
+  useLegacyThemes?: boolean
+  configFilePath: string
+  rootDir: string
+}
+
+interface ILoadedThemes {
+  config: GatsbyConfig
+  themes: Array<IProcessedTheme>
+}
+
+export default async function loadThemes(
+  config: GatsbyConfig,
+  { useLegacyThemes = false, configFilePath, rootDir }: ILoadThemesOptions
+): globalThis.Promise<ILoadedThemes> {
   const themesA = await Promise.mapSeries(
-    useLegacyThemes ? config.__experimentalThemes || [] : config.plugins || [],
+    (useLegacyThemes ? config.__experimentalThemes : config.plugins) || [],
     async themeSpec => {
       const themeObj = await resolveTheme(
```

Comment:
Can you type the return for this please

Reply:
Hey @ascorbic, I typed this function but there're other arrow functions that do not have an explicit return type. Should I add it to these as well?

Reply:
In general, yes.

Reply:
Done